Doc/library/typing.rst

```diff
@@ -1078,6 +1080,24 @@ These are not used in annotations. They are building blocks for creating generic
         ``ParamSpec`` and ``Concatenate``).
       * :class:`Callable` and :class:`Concatenate`.
 
+.. data:: ParamSpecArgs
+.. data:: ParamSpecKwargs
+
+   Arguments and keyword arguments attributes of a :class:`ParamSpec`. The
+   ``P.args`` attribute of a ``ParamSpec`` is an instance of ``ParamSpecArgs``,
+   and ``P.kwargs`` is an instance of ``ParamSpecKwargs``. They are intended
+   for runtime introspection and have no special meaning to static type checkers.
+
+   Calling :func:`get_origin` on either of these objects will return the
+   original ``ParamSpec``::
+
+      P = ParamSpec("P")
+      assert get_origin(P.args) is P
+      assert get_origin(P.kwargs) is P
```

Comment:
While I know the idiom is commonly used and technically correct, formulating an equivalence using an `assert` does not feel clear to me. I'd much rather show it in a REPL fragment, state the equivalency in English, or show it in a comment, e.g.
```
get_origin(P.args)  # returns P
```